src/transformers/configuration_utils.py

```diff
@@ -1050,6 +1069,15 @@ def register_for_auto_class(cls, auto_class="AutoConfig"):
 
         cls._auto_class = auto_class
 
+    def has_set_generation_parameters(self) -> bool:
```

Comment:
Perhaps we can change the name here - as it will return False as long as the generation params passed are default e.g. `def _overrides_generation_parameters`, `_has_non_default_generation_params`

i.e. it can still have 'set' generation params